### PR TITLE
feature(schedule): add status tabs 

### PR DESCRIPTION
**Pull request by nataliagulko**
#### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Performance optimization
- [x] Refactoring
- [ ] Test Case
- [ ] Documentation update
- [ ] Other

#### 🔗 Related issue link

[Schedule: Tabs menu](https://github.com/rolling-scopes/rsschool-app/issues/1736)

#### 💡 Background and solution

- Added `<StatusTabs />` component on Schedule page
- Statuses in table are filtered by click on tabs 
- Added tests for `<StatusTabs />` and `<TableView />` components

##### "All" selected by default

image

##### When tab does not have items

image

##### Specific tab with items

image

#### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Database migration is added or not needed
- [x] Documentation is updated/provided or not needed
- [x] Changes are tested locally

### CODE REVIEW

File: client/src/api/api.ts [hunk 1]
```diff
@@ -766,10 +766,10 @@ export type CourseEventDtoTypeEnum = typeof CourseEventDtoTypeEnum[keyof typeof
 export interface CourseScheduleItemDto {
     /**
      * 
-     * @type {object}
```
Comment (nataliagulko) [inline client/src/api/api.ts:769]: Difference in types were found when I added tests. If this file is auto-generated, how to change types properly?

File: client/src/modules/Schedule/components/StatusTabs/StatusTabs.test.tsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
+import { fireEvent, render, screen } from '@testing-library/react';
```
Comment (nataliagulko) [inline client/src/modules/Schedule/components/StatusTabs/StatusTabs.test.tsx:1]: What is the best practices in this repository to place tests for components? Please guide me

File: client/src/modules/Schedule/pages/SchedulePage/index.tsx [hunk 3]
```diff
@@ -23,6 +25,7 @@ export function SchedulePage(props: PageProps) {
   const [cipher, setCipher] = useState('');
   const [courseTask, setCourseTask] = useState<null | Record<string, any>>(null);
   const [copyModal, setCopyModal] = useState<{ id?: number } | null>(null);
+  const [selectedTab, setSelectedTab] = useState<string>(ALL_TAB_KEY);
```
Comment (nataliagulko) [inline client/src/modules/Schedule/pages/SchedulePage/index.tsx:28]: `statusFilter` at `<TableView />` used `useLocalStorage` hook. I move it from component level to page level and changed to `useState`. Should the filtering via tabs also use `useLocalStorage`?

File: client/src/modules/Schedule/components/StatusTabs/StatusTabs.test.tsx (hunk 2 above)
Comment (apalchys) [reply client/src/modules/Schedule/components/StatusTabs/StatusTabs.test.tsx:1]: @nataliagulko it's fine what you did. We have no guides. If you are ready to define some, please do and for example put it to contribution guide :)

**Review by apalchys — APPROVED** [review]
👍

File: client/src/modules/Schedule/pages/SchedulePage/index.tsx (hunk 3 above)
Comment (apalchys) [reply client/src/modules/Schedule/pages/SchedulePage/index.tsx:28]: I would say - yes

**Comment by apalchys** [comment]
Oops, I miseed you comment about types in `api`. This file is fully auto generated. There is a command in `nestjs`  folder available: `npm run openapi`. To fix this issue we need to fix api definition for models in nestjs. I will show you how it a separate commit.

**Comment by apalchys** [comment]
@nataliagulko This PR shows how to fix types usually:
https://github.com/rolling-scopes/rsschool-app/pull/1748

There is an issue with generator that it does detect correctly union types and mark them as object. In this case, we need to specify them explicitly.

File: client/src/modules/Schedule/pages/SchedulePage/index.tsx (hunk 3 above)
Comment (nataliagulko) [reply client/src/modules/Schedule/pages/SchedulePage/index.tsx:28]: Ok. I will fix and update the PR

Comment (nataliagulko) [reply client/src/modules/Schedule/pages/SchedulePage/index.tsx]: Fixed, please review

File: client/src/modules/Schedule/components/StatusTabs/StatusTabs.test.tsx (hunk 2 above)
Comment (nataliagulko) [reply client/src/modules/Schedule/components/StatusTabs/StatusTabs.test.tsx:1]: Not ready yet, I think. I saw that some other contributors created `__tests__` folders on components level. I would want to discuss it with other developers to get an agreement

**Review by valerydluski — APPROVED** [review]

**Pull request merged**